src/inmanta/deploy/work.py

```diff
@@ -345,7 +347,22 @@ def deploy_with_context(
         for resourceId in queued:
             if resourceId in resources:
                 task = tasks.Deploy(resource=resourceId)
-                self.agent_queues._in_progress[task] = min(self.agent_queues._in_progress[task], priority)
+                assert task in self.agent_queues._in_progress
+                actual_task = [e for e in self.agent_queues._in_progress.keys() if e == task][0]
```

Comment:
When a task is in progress, it could not exist in the `task_by_resource`:
![image](https://github.com/user-attachments/assets/40d1aec1-d0fe-474c-b1c3-203cb0c484a3)